internal/controllers/server/tests/create-minimal/01-assert.yaml

```diff
@@ -0,0 +1,6 @@
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: openstack-clouds
+  deletionGracePeriodSeconds: 0
```

Comment:
If you want to assert that it has been deleted you probably want a cel assertion that `deletionTimestamp` != 0.

The real test here is that the subsequent deletion of the namespace succeeds.

Reply:
I want to assert that it has been flagged for deletion, but not deleted (due to the finalizer). This is the easiest check I could think of.

Reply:
That's exactly what `deletionTimestamp` does. You can tell it hasn't been deleted because it still exists. You can tell it has been flagged for deletion because `deletionTimestamp` has been set.

This is how we test this in controllers too, e.g. https://github.com/k-orc/openstack-resource-controller/blob/325956d1b923b0640cdc303593a9a1565774326e/internal/controllers/generic/controller.go#L124-L125

Reply:
Done, this should be a lot more robust, as I also check that we have a finalizer from server.